backend/src/api/mutation.rs

```diff
@@ -242,37 +237,31 @@ impl Mutation {
         realms: Vec<RealmLineageComponent>,
         context: &Context,
     ) -> ApiResult<CreateRealmLineageOutcome> {
-        if context.auth != AuthContext::TrustedExternal {
-            return Err(not_authorized!("only trusted external applications can use this mutation"));
-        }
-
-        if realms.len() == 0 {
-            return Ok(CreateRealmLineageOutcome { num_created: 0 });
-        }
-
-        if context.config.general.reserved_paths().any(|r| realms[0].path_segment == r) {
-            return Err(invalid_input!(key = "realm.path-is-reserved", "path is reserved and cannot be used"));
-        }
+        Realm::create_lineage(realms, context).await
+    }
 
-        let mut parent_path = String::new();
-        let mut num_created = 0;
-        for realm in realms {
-            let sql = "\
-                insert into realms (parent, name, path_segment) \
-                values ((select id from realms where full_path = $1), $2, $3) \
-                on conflict do nothing";
-            let res = context.db.execute(sql, &[&parent_path, &realm.name, &realm.path_segment])
-                .await;
-            let affected = map_db_err!(res, {
-                if constraint == "valid_path" => invalid_input!("path invalid"),
-            })?;
-            num_created += affected as i32;
+    /// Stores series information in Tobira's DB, so it can be mounted without having to be harvested first.
+    async fn announce_series(series: NewSeries, context: &Context) -> ApiResult<Series> {
+        Series::announce(series, context).await
+    }
 
-            parent_path.push('/');
-            parent_path.push_str(&realm.path_segment);
-        }
+    /// Adds a series block to an empty realm, if the series is known to Tobira.
+    async fn add_series_mount_point(
+        series_oc_id: String,
+        target_path: String,
+        context: &Context,
+    ) -> ApiResult<Realm> {
+        Series::add_mount_point(series_oc_id, target_path, context).await
+    }
 
-        Ok(CreateRealmLineageOutcome { num_created })
+    /// Removes the realm in which the series was previously mounted in, if it has no children.
+    /// Otherwise the realm is given the name of its path segment.
```

Comment:
This is missing the removal of the block, and also error conditions. But I agree that writing nice brief docs for this isnt trivial :D two suggestions.

```suggestion
    /// Removes the series block of given series from the given realm. If the
    /// realm has sub-realms and used to derive its name from the block, it is
    /// renamed to its path segment. If the realm has no sub-realms, it is
    /// removed completely. Errors if the given realm does not have exactly one
    /// series block referring to the specified series.
```

Or closer to yours:

```suggestion
    /// Removes the given realm in which the given series was previously mounted
    /// in, if it has no sub-realms. Otherwise the series block in the realm is
    /// removed and the realm is given the name of its path segment.Errors if
    /// the given realm does not have exactly one series block referring to the
    /// specified series.
```